Intermittent tests failures

**Describe the bug**
I've noticed that on github **and on my machine** tests are failing sometimes, and I'm not sure why.

The failing tests have been:
- `test_can_change_picto_at_datalayer_level[chromium]` and
- `test_can_change_owner[chromium]`

See the tracetracks below.

**To Reproduce**

Run tests with `make test`. Sometimes it happens on the CI as well.

**Tracebacks**

```
______________________________________________________________ test_can_change_owner[chromium] _______________________________________________________________
[gw1] darwin -- Python 3.11.6 /Users/alexis/dev/umap/.venv/bin/python3

map = <Map: test map>, live_server = <LiveServer listening at http://localhost:54480>, login = <function login.<locals>.do_login at 0x11ab92700>
user = <User: Joe>

    def test_can_change_owner(map, live_server, login, user):
        page = login(map.owner)
        page.goto(f"{live_server.url}{map.get_absolute_url()}?edit")
        edit_permissions = page.get_by_title("Update permissions and editors")
        edit_permissions.click()
        close = page.locator(".umap-field-owner .close")
        close.click()
        input = page.locator("input.edit-owner")
        input.type(user.username)
        input.press("Tab")
        save = page.get_by_role("button", name="Save")
        expect(save).to_be_visible()
        save.click()
        sleep(1)  # Let save ajax go
        modified = Map.objects.get(pk=map.pk)
>       assert modified.owner == user
E       assert <User: Gabriel> == <User: Joe>
E         +<User: Gabriel>
E         -<User: Joe>

umap/tests/integration/test_owned_map.py:234: AssertionError
```

and 

```
_____________________________________________________ test_can_change_picto_at_datalayer_level[chromium] _____________________________________________________
[gw0] darwin -- Python 3.11.6 /Users/alexis/dev/umap/.venv/bin/python3

map = <Map: test map>, live_server = <LiveServer listening at http://localhost:54483>
page = <Page url='http://localhost:54483/en/map/test-map_28#7/48.944/13.447'>, pictos = None

    def test_can_change_picto_at_datalayer_level(map, live_server, page, pictos):
        # Faster than doing a login
        map.edit_status = Map.ANONYMOUS
        map.settings["properties"]["iconUrl"] = "/uploads/pictogram/star.svg"
        map.save()
        DataLayerFactory(map=map, data=DATALAYER_DATA)
        page.goto(f"{live_server.url}{map.get_absolute_url()}?edit")
        marker = page.locator(".umap-div-icon img")
        expect(marker).to_have_count(1)
        # Should have default img
        expect(marker).to_have_attribute("src", "/uploads/pictogram/star.svg")
        # Edit datalayer
        marker.click(modifiers=["Control", "Shift"])
        settings = page.get_by_text("Layer properties")
>       expect(settings).to_be_visible()
E       AssertionError: Locator expected to be visible
E       Actual value: None
E       Call log:
E       LocatorAssertions.to_be_visible with timeout 5000ms
E       waiting for get_by_text("Layer properties")

umap/tests/integration/test_picto.py:85: AssertionError
```

--- a/umap/tests/integration/test_owned_map.py
+++ b/umap/tests/integration/test_owned_map.py
@@ -1,4 +1,4 @@
-from time import sleep
+import re
 
 import pytest
 from playwright.sync_api import expect
@@ -20,7 +20,6 @@ def login(context, settings, live_server):
         page.get_by_placeholder("Username").fill(user.username)
         page.get_by_placeholder("Password").fill("123123")
         page.locator('#login_form input[type="submit"]').click()
-        sleep(1)  # Time for ajax login POST to proceed
         return page
 
     return do_login
@@ -183,8 +182,8 @@ def test_create(tilelayer, live_server, login, user):
     expect(marker).to_have_count(1)
     save = page.get_by_role("button", name="Save")
     expect(save).to_be_visible()
-    save.click()
-    sleep(1)  # Let save ajax go back
+    with page.expect_response(re.compile(r".*/datalayer/create/")):
+        save.click()
     expect(marker).to_have_count(1)
 
 
@@ -197,8 +196,8 @@ def test_can_change_perms_after_create(tilelayer, live_server, login, user):
     page.locator("input[name=name]").fill("Layer 1")
     save = page.get_by_role("button", name="Save")
     expect(save).to_be_visible()
-    save.click()
-    sleep(1)  # Let save ajax go back
+    with page.expect_response(re.compile(r".*/map/create/")):
+        save.click()
     edit_permissions = page.get_by_title("Update permissions and editors")
     expect(edit_permissions).to_be_visible()
     edit_permissions.click()
@@ -232,7 +231,7 @@ def test_can_change_owner(map, live_server, login, user):
     input.press("Tab")
     save = page.get_by_role("button", name="Save")
     expect(save).to_be_visible()
-    save.click()
-    sleep(1)  # Let save ajax go
+    with page.expect_response(re.compile(r".*/update/permissions/.*")):
+        save.click()
     modified = Map.objects.get(pk=map.pk)
     assert modified.owner == user

